src/Forms/FieldList.php

```diff
@@ -58,6 +58,38 @@ public function __clone()
         }
     }
 
+    /**
+     * Iterate over each field in the current list recursively
+     *
+     * @param callable $callback
+     */
+    public function recursiveWalk(callable $callback)
+    {
+        $stack = $this->toArray();
+        while (!empty($stack)) {
+            /** @var FormField $field */
+            $field = array_shift($stack);
+            $callback($field);
+            if ($field instanceof CompositeField) {
+                $stack = array_merge($field->getChildren()->toArray(), $stack);
+            }
+        }
+    }
+
+    /**
+     * Return a flattened array of all fields
+     *
+     * @return static
+     */
+    public function flattenFields()
+    {
+        $fields = [];
+        $this->recursiveWalk(function (FormField $field) use (&$fields) {
+            $fields[] = $field;
```

Comment:
If we are flattening fields, should we skip composite fields? E.g. merge in fields in nested FieldGroups, but not the FieldGroup themselves?

Reply:
Not for flattening explicitly,  don't think... it is a field regardless of it's type. I'd much rather `flattenFields` be agnostic about this point...